test/test_dht.cpp

```diff
@@ -508,10 +508,11 @@ void do_test_dht(address(&rand_addr)())
 	mock_socket s;
 	obs observer;
 	counters cnt;
+	boost::scoped_ptr<dht_storage_interface> dht_storage(dht_default_storage_constructor(sett));
```

Comment:
I would prefer std::unique_ptr over boost::scoped_ptr.. but I will have to replace them all in one big patch at some point